asv_bench/benchmarks/inference.py

```diff
@@ -1,117 +1,114 @@
 
-    def time_int64(self):
-        (self.df_int64['A'] + self.df_int64['B'])
+    def time_subtract(self, dtype):
+        self.df['A'] - self.df['B']
 
-    def time_int32(self):
-        (self.df_int32['A'] + self.df_int32['B'])
+    def time_multiply(self, dtype):
+        self.df['A'] * self.df['B']
 
-    def time_uint32(self):
-        (self.df_uint32['A'] + self.df_uint32['B'])
+    def time_divide(self, dtype):
+        self.df['A'] / self.df['B']
 
-    def time_float64(self):
-        (self.df_float64['A'] + self.df_float64['B'])
+    def time_modulo(self, dtype):
+        self.df['A'] % self.df['B']
 
-    def time_float32(self):
-        (self.df_float32['A'] + self.df_float32['B'])
 
-    def time_datetime64(self):
-        (self.df_datetime64['A'] - self.df_datetime64['B'])
+class DateInferOps(object):
+    # from GH 7332
+    goal_time = 0.2
+
+    def setup_cache(self):
+        N = 5 * 10**5
+        df = DataFrame({'datetime64': np.arange(N).astype('datetime64[ms]')})
+        df['timedelta'] = df['datetime64'] - df['datetime64']
+        return df
 
-    def time_timedelta64_1(self):
-        (self.df_timedelta64['A'] + self.df_timedelta64['B'])
+    def time_subtract_datetimes(self, df):
+        df['datetime64'] - df['datetime64']
 
-    def time_timedelta64_2(self):
-        (self.df_timedelta64['A'] + self.df_timedelta64['A'])
+    def time_timedelta_plus_datetime(self, df):
+        df['timedelta'] + df['datetime64']
 
+    def time_add_timedeltas(self, df):
+        df['timedelta'] + df['timedelta']
+
+
+class ToNumeric(object):
 
-class to_numeric(object):
     goal_time = 0.2
+    params = ['ignore', 'coerce']
+    param_names = ['errors']
 
-    def setup(self):
-        self.n = 10000
-        self.float = Series(np.random.randn(self.n * 100))
+    def setup(self, errors):
+        N = 10000
+        self.float = Series(np.random.randn(N))
         self.numstr = self.float.astype('str')
-        self.str = Series(tm.makeStringIndex(self.n))
-
-    def time_from_float(self):
-        pd.to_numeric(self.float)
+        self.str = Series(tm.makeStringIndex(N))
 
-    def time_from_numeric_str(self):
-        pd.to_numeric(self.numstr)
+    def time_from_float(self, errors):
+        to_numeric(self.float, errors=errors)
 
-    def time_from_str_ignore(self):
-        pd.to_numeric(self.str, errors='ignore')
+    def time_from_numeric_str(self, errors):
+        to_numeric(self.numstr, errors=errors)
 
-    def time_from_str_coerce(self):
-        pd.to_numeric(self.str, errors='coerce')
+    def time_from_str(self, errors):
+        to_numeric(self.str, errors=errors)
 
 
-class to_numeric_downcast(object):
+class ToNumericDowncast(object):
 
     param_names = ['dtype', 'downcast']
     params = [['string-float', 'string-int', 'string-nint', 'datetime64',
                'int-list', 'int32'],
               [None, 'integer', 'signed', 'unsigned', 'float']]
 
     N = 500000
-    N2 = int(N / 2)
```

Comment:
this was for a good reason, you can't use floats to multiply lists (I find it a bit strange that this didn't fail for you)